### PR TITLE
feat: add builder methods

### PR DESCRIPTION
**Pull request by EmperorOrokuSaki**
## Motivation
Issue #583 

## Solution
Implement builder methods for all fields

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/rpc-types-trace/src/filter.rs [hunk 1]
```diff
@@ -33,6 +33,48 @@ pub struct TraceFilter {
 // === impl TraceFilter ===
 
 impl TraceFilter {
+    /// Sets the `from_block` field of the struct
+    pub fn with_from_block(&mut self, block: u64) -> &mut Self {
```
Comment (mattsse) [inline crates/rpc-types-trace/src/filter.rs:37]: these should all consume `self` and return `Self`

File: crates/rpc-types-trace/src/filter.rs
Comment (mattsse) [inline crates/rpc-types-trace/src/filter.rs:1]: can drop the `with_` prefix, is cleaner imo

File: crates/rpc-types-trace/src/geth/mod.rs [hunk 2]
```diff
@@ -497,6 +497,26 @@ pub struct GethDebugTracingCallOptions {
...
+    }
+
+    /// Enables block overrides
+    pub fn with_block_overrides(&mut self, overrides: BlockOverrides) -> &mut Self {
+        self.block_overrides = Some(overrides);
+        self
+    }
+
+    /// Sets the tracing options
+    pub fn with_tracing_options(&mut self, options: GethDebugTracingOptions) -> &mut Self {
```
Comment (mattsse) [inline crates/rpc-types-trace/src/geth/mod.rs:514]: here we can keep the with_ but also change fro &mut self to self

**Review by mattsse — CHANGES_REQUESTED** [review]
ty,
some nits

**Review by prestwich — APPROVED** [review]

**Pull request merged**